docker/Dockerfile

```diff
@@ -8,50 +8,115 @@
 
-# Some RUN statements are combined together to make Docker build run faster.
 # Get latest package listing, install software-properties-common, git, wget,
 # compilers and libraries.
 # git is required for pyproject.toml toolchain's use of CMakeLists.txt.
-# gcc, g++, make are required for compiling hmmer and AlphaFold 3 libaries.
+# gcc, g++, make are required for compiling AlphaFold 3 libaries.
 # zlib is a required dependency of AlphaFold 3.
-RUN apt update --quiet \
-    && apt install --yes --quiet software-properties-common \
-    && apt install --yes --quiet git wget gcc g++ make zlib1g-dev zstd
-
-# Get apt repository of specific Python versions. Then install Python. Tell APT
-# this isn't an interactive TTY to avoid timezone prompt when installing.
-RUN add-apt-repository ppa:deadsnakes/ppa \
-    && DEBIAN_FRONTEND=noninteractive apt install --yes --quiet python3.11 python3-pip python3.11-venv python3.11-dev
+# Use deadsnakes/ppa apt repository to install specific Python versions
+RUN apt-get update && apt-get install --no-install-recommends -y \
+        software-properties-common \
+    && add-apt-repository ppa:deadsnakes/ppa \
+    && apt-get update && apt-get install --no-install-recommends -y \
+        git \
+        wget \
+        gcc \
+        g++ \
+        make \
+        zlib1g-dev \
+        zstd \
+        python3.11 \
+        python3-pip \
+        python3.11-venv \
+        python3.11-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Create Python 3.11 virtual environment for the AlphaFold build
 RUN python3.11 -m venv /alphafold3_venv
-ENV PATH="/hmmer/bin:/alphafold3_venv/bin:$PATH"
+
 # Update pip to the latest version. Not necessary in Docker, but good to do when
 # this is used as a recipe for local installation since we rely on new pip
 # features for secure installs.
-RUN pip3 install --upgrade pip
-
-# Install HMMER. Do so before copying the source code, so that docker can cache
-# the image layer containing HMMER.
-RUN mkdir /hmmer_build /hmmer ; \
-    wget http://eddylab.org/software/hmmer/hmmer-3.4.tar.gz --directory-prefix /hmmer_build ; \
-    (cd /hmmer_build && tar zxf hmmer-3.4.tar.gz && rm hmmer-3.4.tar.gz) ; \
-    (cd /hmmer_build/hmmer-3.4 && ./configure --prefix /hmmer) ; \
-    (cd /hmmer_build/hmmer-3.4 && make -j8) ; \
-    (cd /hmmer_build/hmmer-3.4 && make install) ; \
-    (cd /hmmer_build/hmmer-3.4/easel && make install) ; \
-    rm -R /hmmer_build
-
-# Copy the AlphaFold 3 source code from the local machine to the container and
+RUN /alphafold3_venv/bin/pip3 install --no-cache-dir --upgrade pip
+
+# Add the AlphaFold 3 source code from the local machine to the container and
 # set the working directory to there.
-COPY . /app/alphafold
+# COPY . /app/alphafold
+
+# Add the AlphaFold 3 source code from GitHub to the image and
+# set the working directory to there
+ADD https://github.com/google-deepmind/alphafold3.git#v${ALPHAFOLD_VERSION} /app/alphafold
```

Comment:
This takes a different approach to the existing Dockerfile (which copies the contents from the local repository) and instead allows the Docker build to act as a stand-alone process.